distributed_pcg.cpp

```diff
@@ -109,16 +90,33 @@ class MapMatrix {
     for (int i=0; i < rows; i++) {
       // this is k=row_ptrs[i] to row_ptrs[i+1] - 1
       for (int k=row_ptrs[i]; k < row_ptrs[i+1]; k++) {
-        result[i] += values[k] * xi[col_idxs[k]]; 
+        result[i] += V[k] * xi[col_idxs[k]]; 
         // k will index into the values list because row_ptrs holds the indices of the values in V in each row 
         // it also works for col indicies (k-th nonzero element)
       }
     }
     return result; 
+  }
 
-
-
+  void print()
+  {
+    std::cout << "Rows: " << rows << std::endl;
+    std::cout << "Cols: " << cols << std::endl;
+    std::cout << "num_values: " << V.size() << std::endl;
+    std::cout << "First col_idx: " << col_idxs[0] << std::endl;
+    std::cout << "row_ptrs.size(): " << row_ptrs.size() << std::endl;
+
+      for (int i = 0; i < row_ptrs.size() - 1; i++)  // Ensure we don't go out of bounds
```

Comment:
Note: Upon entering this line, if I recall correctly, we got an error, connected to the fact that row_ptrs.size() is zero

I think Jaycee did work to fix this but I haven't looked at that yet: maybe next action item for me is to merge her work into this branch